sklearn/preprocessing/tests/test_polynomial.py

```diff
@@ -552,6 +552,39 @@ def test_polynomial_features_csr_X(deg, include_bias, interaction_only, dtype):
     assert_array_almost_equal(Xt_csr.A, Xt_dense)
 
 
+@pytest.mark.parametrize("columns", [1, 2, 3, 1000])
+def test_polynomial_features_csr_wide(columns):
```

Comment:
What are we trying to test here? That we don't get a crash for wide data? Do we currently get a crash in more cases than after this pull request?

Reply:
This is basically a test for https://github.com/scikit-learn/scikit-learn/issues/16803 and as such, maybe shouldn't be in this PR.

It also serves as a kind of performance regression test since it will run very slowly on the old version of `fit`. Are there actual performance tests anywhere?

Reply:
we have the ASV benchmarks but no other regular performance tests.